README_LLaMA_BYOCTritonTRTLLM.md

```diff
@@ -0,0 +1,148 @@
+## Using Triton TensorRT-LLM Backend through building the container
+
+### 1. Get TRTLLM Backend repo
+```
+git clone https://github.com/triton-inference-server/tensorrtllm_backend.git -b v0.7.1
+cd tensorrtllm_backend
+git lfs install
+git submodule update --init --recursive
+```
+
+### 2. Build the Triton TRT-LLM backend container (this will also install `tensorrt_llm` automatically) but building the container can take a while.
+
+```
+DOCKER_BUILDKIT=1 docker build -t triton_trt_llm -f dockerfile/Dockerfile.trt_llm_backend .
+```
+
+### 3. Launch the docker container
+```
+docker run --rm -it --net host --shm-size=2g --ulimit memlock=-1 --ulimit stack=67108864 --gpus all -v $(pwd):/workspace -w /workspace  triton_trt_llm bash
+```
+
+### 4. Build engines
+```
+cd tensorrt_llm/examples/llama
+
+# Download weights from HuggingFace Transformers
+git lfs install
+git clone https://huggingface.co/meta-llama/Llama-2-7b-hf
+
+# Build TensorRT engines
+export HF_LLAMA_MODEL=Llama-2-7b-hf/
+python build.py --model_dir ${HF_LLAMA_MODEL} \
+                --dtype float16 \
+                --remove_input_padding \
+                --use_gpt_attention_plugin float16 \
+                --enable_context_fmha \
+                --use_gemm_plugin float16 \
+                --output_dir /tmp/llama/7B/trt_engines/fp16/1-gpu/ \
+                --paged_kv_cache \
+                --max_batch_size 64
+                --world_size 4 \
+                --tp_size 4 \
+                --pp_size 1
+```
+
+### 5. Create model repository
+```
+# Create the model repository that will be used by the Triton server
+cd /workspace/tensorrtllm_backend
+mkdir triton_model_repo
+
+# Copy the example models to the model repository
+cp -r all_models/inflight_batcher_llm/ensemble triton_model_repo/
+cp -r all_models/inflight_batcher_llm/preprocessing triton_model_repo/
+cp -r all_models/inflight_batcher_llm/postprocessing triton_model_repo/
+cp -r all_models/inflight_batcher_llm/tensorrt_llm triton_model_repo/
+
+# Prepare configs
+```
+
+### 6. Modify the model configuration
+The following table shows the fields that need to be modified before deployment:
+
+*triton_model_repo/preprocessing/config.pbtxt*
+
+Mandatory ones are
+| Name | Description
+| :----------------------: | :-----------------------------: |
+| `triton_max_batch_size` | Here setting to 4 |

```

Comment:
I see you are setting it to 64 in all the config.pbtxt, can you please update the doc to say "Here setting to 64"